### PR TITLE
Implement `GetClusterAccessGraphConfig` method

### PR DESCRIPTION
**Pull request by tigrato**
This PR implements the `GetClusterAccessGraphConfig` method by reading the config provided to Auth server. There is no persistency and the data is read-only. The source of truth is the config parameter set in Auth server's `teleport.yaml`.

### CODE REVIEW

File: lib/auth/clusterconfig/clusterconfigv1/service_test.go [hunk 1]
```diff
@@ -1755,3 +1764,111 @@ func newTestEnv(opts ...serviceOpt) (*env, error) {
...
+					Address:  "address",
+					Ca:       []byte("ca"),
+				},
+			},
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			if test.testSetup != nil {
```
Comment (jakule) [inline lib/auth/clusterconfig/clusterconfigv1/service_test.go:1853]: nit: I'd remove this check and let the test panic if the setup function is not set. Panic here is less evil than forgetting to set the setup function, even if it's empty.

File: lib/auth/clusterconfig/clusterconfigv1/service_test.go [hunk 2]
```diff
@@ -1755,3 +1764,111 @@ func newTestEnv(opts ...serviceOpt) (*env, error) {
...
+				return authRoleContext, nil
+			})
+			env, err := newTestEnv(withAuthorizer(authorizer), withAccessGraphConfig(test.accessGraphConfig))
+			require.NoError(t, err, "creating test service")
+
+			got, err := env.GetClusterAccessGraphConfig(context.Background(), &clusterconfigpb.GetClusterAccessGraphConfigRequest{})
+			test.errorAssertion(t, err)
+
+			require.Empty(t, cmp.Diff(test.responseAssertion, got, protocmp.Transform()))
+
```
Comment (jakule) [inline lib/auth/clusterconfig/clusterconfigv1/service_test.go:1871]: nit:
```suggestion
```

File: lib/service/service.go [hunk 3]
```diff
@@ -2033,13 +2033,26 @@ func (process *TeleportProcess) initAuthService() error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	var accessGraphCAData []byte
+	if cfg.AccessGraph.Enabled && cfg.AccessGraph.CA != "" {
+		accessGraphCAData, err = os.ReadFile(cfg.AccessGraph.CA)
+		if err != nil {
+			return trace.Wrap(err)
```
Comment (jakule) [inline lib/service/service.go:2040]: nit:
```suggestion
			return trace.WithUserMessage(err, "failed to read access graph CA")
```

File: lib/auth/grpcserver_test.go [hunk 4]
```diff
@@ -4388,3 +4389,66 @@ func TestDropDBClientCAEvents(t *testing.T) {
...
+
+	tests := []struct {
+		desc      string
+		role      types.SystemRole
+		assertErr require.ErrorAssertionFunc
+		expected  *clusterconfigpb.AccessGraphConfig
+	}{
+		{
+			desc:      "admin role can't pull the access graph config",
+			role:      types.RoleAdmin,
```
Comment (jakule) [inline lib/auth/grpcserver_test.go:4423]: I don't think this makes sense, as the admin can still pull the resources from the backend anyway.

**Review by jakule — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**